apps/pushes/views/diff.py

```diff
@@ -29,6 +29,9 @@ class BadRevision(Exception):
 
 
 class DiffView(View):
+    moved = {}
+    copied = {}
```

Comment:
How did this work before this addition? Is it necessary given that line 131 also initializes these dicts?

Reply:
This wasn't used before, as I wasn't unit-testing this view at all, basically. All tests went through the initial .get(), which set these. Having these as dicts on the class isn't great, but it's not much of a problem. In practice, in each view, a new instance is instantiated, which overwrites the class ones. I also do that in tests, so the class-members ones are only ever read from, and in their empty state.

Alternatively I'd overwrite the constructor, but the default one is kind nice and generic, didn't want to touch that.

I'll add a comment.

Reply:
... actually, trivial enough to subclass dict to not be mutable, so I did that.